nilearn/input_data/tests/test_multi_nifti_masker.py

```diff
@@ -126,3 +127,61 @@ def test_joblib_cache():
         mask_hash = hash(masker.mask_img_)
         masker.mask_img_.get_data()
         assert_true(mask_hash == hash(masker.mask_img_))
+
+
+def test_transform_cache():
```

Comment:
Also, I think that you could probably factor a bit this tests together: they contain a lot of duplicated code.

If you want you can create an auxiliary function that, given a masker and niimgs, fits and transforms thoses niimgs and raises an error if there is a recomputation.